packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DraggingCard/DraggingCard.tsx

```diff
@@ -0,0 +1,179 @@
+    this.cardWidth = 374;
+    this.cardHeight = 296;
+  }
+
+  public render(): JSX.Element {
+    let className = mergeStyles(getStyles().container, {
+      width: this.cardWidth + 'px',
+      height: this.cardHeight + 'px'
+    });
+    className = 'draggingCardDGL' + ' ' + className;
+    const animation = this.mapAnimationToRender(this.props.draggingAnimation);
+    return (
+      <div className={className}>
+        <div className={this.classNames.title}>
+          <div className={this.classNames.titleText}> {this.props.title}</div>
+        </div>
+        <div className={this.classNames.body}>{animation}</div>
+      </div>
+    );
+  }
+
+  private mapAnimationToRender = (draggingAnimation?: DraggingAnimationType) => {
+    switch (draggingAnimation) {
+      case DraggingAnimationType.BarGraph: {
+        return <BarGraph />;
+      }
+      case DraggingAnimationType.DonutChart: {
+        return <DonutChart />;
+      }
+      case DraggingAnimationType.HorizontalBarGraph: {
+        return <HorizontalBarGraph />;
+      }
+      case DraggingAnimationType.LineChart: {
+        return <LineChart />;
+      }
+      case DraggingAnimationType.Shimmer: {
+        return <Shimmer />;
+      }
+    }
+  };
+
+  private moveHandler = (event: MouseEvent) => {
+    // check if the dragging card has been brought too close to add card panel, if so stop rendering dragging card and place the card back
+    // used to place the dragging card back in the add card panel
+    if (event.clientX - this.props.initialX > 400) {
+      window.document.removeEventListener('mouseup', this.stopPlaceholder);
+      window.document.removeEventListener('mousemove', this.moveHandler);
+      this.props.setDragMode();
+      return;
+    }
+
+    // Change left and top styling on the dragging card so that it is always next to cursor
+    const draggingCardDomElement = document.getElementsByClassName('draggingCardDGL');
+    if (draggingCardDomElement && draggingCardDomElement[0]) {
+      (draggingCardDomElement[0] as HTMLElement).style.left = event.clientX - this.cardWidth / 2 + 'px';
+      (draggingCardDomElement[0] as HTMLElement).style.top = event.clientY - this.cardHeight / 2 + 'px';
+    }
+    if (dragApi && dragApi.value) {
+      const containerRectList = document.getElementsByClassName('dashboardContainerClassName');
+      const containerRect = containerRectList[0] ? containerRectList[0] : null;
+      const width = this.props.cardSize === CardSize.mediumWide || this.props.cardSize === CardSize.large ? 2 : 1;
+      const height = this.props.cardSize === CardSize.small || this.props.cardSize === CardSize.mediumWide ? 4 : 8;
+      if (containerRect) {
+        const boundingRect = containerRect.getBoundingClientRect();
+        const left = event.clientX - boundingRect.left;
+        const right = boundingRect.right - event.clientX;
+        const top = event.clientY - boundingRect.top;
+        if (left < 0 || top < 0 || right < 0) {
```

Comment:
>    if (left < 0 || top < 0 || right < 0)  [](start = 4, length = 42)

this may change based on rtl or ltr, have you tested that scenario?

Reply:
Tested it. The boundary conditions stay the same